src/Reader.php

```diff
@@ -400,7 +400,8 @@ private function buildSheetImports($import): array
                 method_exists($this->reader, 'setLoadSheetsOnly')
                 && count(array_filter(array_keys($sheetImports), 'is_numeric')) === 0
             ) {
-                $this->reader->setLoadSheetsOnly(array_keys($sheetImports));
+                // @todo PHPOffice v2 upgrade
+//                $this->reader->setLoadSheetsOnly(array_keys($sheetImports));
```

Comment:
So this is only a problem when the user provided invalid sheet names?

Reply:
That’s right - if user provides an invalid sheet name, the whole load method fails and we have to attempt to load it again with all sheets, as proposed in the diff above which doesn’t seem great either.
I wonder what’s the memory footprint difference so I’ll do some benchmarks to see but loading the file again might not be the best way to do it.

Reply:
Hm, maybe we can somehow get the sheets names cheaply so we can diff them and drop the ones that don't exist. Perhaps a concern that throws and error if trying to load non existing sheets so it can be opt-in.